flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetInputFormat.java

```diff
@@ -97,11 +97,13 @@
      */
     protected ParquetInputFormat(Path path, MessageType messageType) {
         super(path);
-        this.expectedFileSchema = checkNotNull(messageType, "messageType");
-        RowTypeInfo rowTypeInfo =
-                (RowTypeInfo) ParquetSchemaConverter.fromParquetType(expectedFileSchema);
-        this.fieldTypes = rowTypeInfo.getFieldTypes();
-        this.fieldNames = rowTypeInfo.getFieldNames();
+        this.expectedFileSchema = messageType;
```

Comment:
Please annotate the nullable fields no with `@Nullable` and add a comment like, `can be null if it should be inferred from the file` or so.

Reply:
done